app/controllers/devices_controller.rb

```diff
@@ -0,0 +1,36 @@
+class DevicesController < ApplicationController
+  before_action :authenticate_user!, only: [:create]
+
+  rescue_from ActiveRecord::ActiveRecordError do |exc|
+    render json: { error: exc.message, status: 422 }, status: :unprocessable_entity
+  end
+
+  def create
+    device = Device.find_or_create_by(device_params)
+    if device.persisted?
+      head :ok
+    else
+      render json: { error: device.errors_as_sentence }, status: :bad_request
+    end
+  end
+
+  def destroy
+    device = Device.where(user_id: params[:id],
+                          token: params[:token],
+                          platform: params[:platform],
+                          app_bundle: params[:app_bundle]).first
```

Comment:
Usually `Model.where(stuff).first` can be simplified into `Model.find_by(stuff)`. Also, is there any reason not to reuse `device_params` here?

```suggestion
    device = Device.find_by(device_params)
```

At first I thought this might subject us to an identifier-injection vulnerability, but an attacker would have to know the token and everything else, which would make that virtually impossible to exploit. Either way, I think reusing `device_params` here might make sense.

Reply:
✅